Px.Utils/Validation/SyntaxValidation/SyntaxValidationUtilityMethods.cs

```diff
@@ -0,0 +1,404 @@
+        /// <returns>Returns a boolean which is true if the search was successful. If the search times out, false is returned.</returns>
+        public static bool FindIllegalCharactersInString(string input, char[] illegalCharacters, out char[] foundIllegalCharacters, int customTimeout = DEFAULT_TIMEOUT)
+        {
+            string processedCharacters = ProcessCharsForRegexPattern(illegalCharacters);
+            string pattern = $"[{processedCharacters}]";
+
+            TimeSpan timeout = TimeSpan.FromSeconds(customTimeout);
+            try
+            {
+                MatchCollection matches = Regex.Matches(input, pattern, RegexOptions.Singleline, timeout);
+                foundIllegalCharacters = matches.Cast<Match>().Select(m => m.Value[0]).ToArray();
+                return true;
+            }
+            catch (RegexMatchTimeoutException)
+            {
+                foundIllegalCharacters = [];
+                return false;
+            }
+        }
+        private static int GetNextLineChangeIndex(string input, PxFileSyntaxConf syntaxConf, int startIndex = 0)
+        {
+            int lineBreakIndex = input.IndexOf(syntaxConf.Symbols.Linebreak, startIndex);
+            int carriageReturnIndex = input.IndexOf(syntaxConf.Symbols.CarriageReturn, startIndex);
+
+            // If neither character was found, return -1
+            if (lineBreakIndex == -1 && carriageReturnIndex == -1)
+            {
+                return -1;
+            }
+
+            // If only one character was found, return its index
+            if (lineBreakIndex == -1)
+            {
+                return carriageReturnIndex;
+            }
+            if (carriageReturnIndex == -1)
+            {
+                return lineBreakIndex;
+            }
+
+            // If both characters were found, return the index of the one that appears first
+            return Math.Min(lineBreakIndex, carriageReturnIndex);
+        }
+
+        private static string ProcessCharsForRegexPattern(char[] illegalCharacters)
+        {
+            string[] specialCharacters = [ "\\", "^", "$", ".", "|", "?", "*", "+", "(", ")", "[", "]", "[", "]" ];
+            string[] processedCharacters = new string[illegalCharacters.Length];
+
+            for (int i = 0; i < illegalCharacters.Length; i++)
+            {
+                string character = illegalCharacters[i].ToString();
+
+                if (specialCharacters.Contains(character))
+                {
+                    // If the character is a special character, escape it
+                    character = "\\" + character;
+                }
+
+                processedCharacters[i] = character;
+            }
+
+            return string.Join("", processedCharacters);
+        }
+
+        private static int FindSymbolIndex(string remainder, char symbol, Dictionary<int, int> stringIndeces, int startSearchIndex = 0)
```

Comment:
Either document the parameters to make idea easier to understand or change the implementation to use simillar algorithm that is used in the syntax validation method.